server/tserver/src/main/java/org/apache/accumulo/tserver/log/SortedLogRecovery.java

```diff
@@ -16,132 +16,79 @@
  */
 package org.apache.accumulo.tserver.log;
 
+import static com.google.common.base.Preconditions.checkState;
+import static org.apache.accumulo.tserver.log.RecoveryLogsIterator.maxKey;
+import static org.apache.accumulo.tserver.log.RecoveryLogsIterator.minKey;
 import static org.apache.accumulo.tserver.logger.LogEvents.COMPACTION_FINISH;
 import static org.apache.accumulo.tserver.logger.LogEvents.COMPACTION_START;
 import static org.apache.accumulo.tserver.logger.LogEvents.DEFINE_TABLET;
 import static org.apache.accumulo.tserver.logger.LogEvents.MANY_MUTATIONS;
 import static org.apache.accumulo.tserver.logger.LogEvents.MUTATION;
-import static org.apache.accumulo.tserver.logger.LogEvents.OPEN;
 
 import java.io.IOException;
+import java.util.Collection;
 import java.util.HashSet;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Map.Entry;
 import java.util.Set;
 
 import org.apache.accumulo.core.data.Mutation;
 import org.apache.accumulo.core.data.impl.KeyExtent;
 import org.apache.accumulo.core.metadata.RootTable;
 import org.apache.accumulo.server.fs.VolumeManager;
+import org.apache.accumulo.tserver.logger.LogEvents;
 import org.apache.accumulo.tserver.logger.LogFileKey;
 import org.apache.accumulo.tserver.logger.LogFileValue;
 import org.apache.hadoop.fs.Path;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.google.common.base.Function;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.PeekingIterator;
+
 /**
  * Extract Mutations for a tablet from a set of logs that have been sorted by operation and tablet.
  *
  */
 public class SortedLogRecovery {
-  private static final Logger log = LoggerFactory.getLogger(SortedLogRecovery.class);
-
-  static class EmptyMapFileException extends Exception {
-    private static final long serialVersionUID = 1L;
-
-    public EmptyMapFileException() {
-      super();
-    }
-  }
 
-  static class UnusedException extends Exception {
-    private static final long serialVersionUID = 1L;
-
-    public UnusedException() {
-      super();
-    }
-  }
+  private static final Logger log = LoggerFactory.getLogger(SortedLogRecovery.class);
 
   private VolumeManager fs;
 
   public SortedLogRecovery(VolumeManager fs) {
     this.fs = fs;
   }
 
-  private enum Status {
-    INITIAL, LOOKING_FOR_FINISH, COMPLETE
-  }
-
-  private static class LastStartToFinish {
-    long lastStart = -1;
-    long seq = -1;
-    long lastFinish = -1;
-    Status compactionStatus = Status.INITIAL;
-    String tserverSession = "";
-
-    private void update(long newFinish) {
-      this.seq = this.lastStart;
-      if (newFinish != -1)
-        lastFinish = newFinish;
-    }
+  private int findMaxTabletId(KeyExtent extent, List<Path> recoveryLogs) throws IOException {
+    int tid = -1;
```

Comment:
Should use less abstract name than "tid".  We use this name for different things throughout the code.